core/src/androidTest/java/org/hisp/dhis/android/core/dataset/internal/DataSetCompleteRegistrationPostCallMockIntegrationShould.kt

```diff
@@ -0,0 +1,139 @@
+
+    @After
+    @Throws(D2Error::class)
+    fun tearDown() {
+        d2.wipeModule().wipeData()
+    }
+
+    @Test
+    fun post_dataset_complete_registration_success() {
+        // Given a DatasetComplete registration to post
+        dhis2MockServer.enqueueMockResponse(200, "imports/data_value_import_summary_web_response.json")
+        val dataSetCompleteRegistration = provideDataSetCompleteRegistration()
+        d2.dataSetModule().dataSetCompleteRegistrations().value(
+            period = dataSetCompleteRegistration.period(),
+            organisationUnit = dataSetCompleteRegistration.organisationUnit(),
+            dataSet = dataSetCompleteRegistration.dataSet(),
+            attributeOptionCombo = dataSetCompleteRegistration.attributeOptionCombo()
+        ).blockingSet()
+
+        val toPostRepository = d2.dataSetModule()
+            .dataSetCompleteRegistrations()
+            .bySyncState().eq(State.TO_POST)
+
+        val syncedRepository = d2.dataSetModule()
+            .dataSetCompleteRegistrations()
+            .bySyncState().eq(State.SYNCED)
+
+        assertEquals(toPostRepository.blockingGet().size, 1)
+        assertEquals(syncedRepository.blockingGet().size, 0)
+
+        // When the object is uploaded
+        d2.dataSetModule().dataSetCompleteRegistrations().blockingUpload()
+
+        // Then the status is synced
+        assertEquals(toPostRepository.blockingGet().size, 0)
+        assertEquals(syncedRepository.blockingGet().size, 1)
+    }
+
+    @Test
+    fun post_dataset_delete_complete_registration_success() {
+        val toPostRepository = d2.dataSetModule()
+            .dataSetCompleteRegistrations()
+            .bySyncState().eq(State.TO_POST)
+
+        val syncedRepository = d2.dataSetModule()
+            .dataSetCompleteRegistrations()
+            .bySyncState().eq(State.SYNCED)
+
+        val toUpdateRepository = d2.dataSetModule()
+            .dataSetCompleteRegistrations()
+            .bySyncState().eq(State.TO_UPDATE)
+
+        val dataSetCompleteRegistration = provideDataSetCompleteRegistration()
+        var dObject = d2.dataSetModule().dataSetCompleteRegistrations().value(
+            period = dataSetCompleteRegistration.period(),
+            organisationUnit = dataSetCompleteRegistration.organisationUnit(),
+            dataSet = dataSetCompleteRegistration.dataSet(),
+            attributeOptionCombo = dataSetCompleteRegistration.attributeOptionCombo()
+        )
+
+        dhis2MockServer.enqueueMockResponse(200, "imports/data_value_import_summary_web_response.json")
+        dObject.blockingSet()
+        assertEquals(toPostRepository.blockingGet().size, 1)
+        assertEquals(syncedRepository.blockingGet().size, 0)
+
+        d2.dataSetModule().dataSetCompleteRegistrations().blockingUpload()
+
+        assertEquals(toPostRepository.blockingGet().size, 0)
+        assertEquals(syncedRepository.blockingGet().size, 1)
+
+        dhis2MockServer.enqueueMockResponse(200)
```

Comment:
Checking the integration with v40, the response is 204 instead of 200. I am not sure if it was 200 before, or if it has been 204 for a while. It shouldn't change anything because it is a success code, but it is better to have it aligned and documented in the test
![image](https://github.com/dhis2/dhis2-android-sdk/assets/6523240/8ede7249-536e-479b-80ff-ad897744ab47)